diktat-maven-plugin/src/test/kotlin/org/cqfn/diktat/plugin/maven/DiktatMavenPluginIntegrationTest.kt

```diff
@@ -5,60 +5,105 @@ import com.soebes.itf.jupiter.extension.MavenJupiterExtension
+ * phase_ for the `diktat-maven-plugin` submodule, as if you were running
+ *
+ * ```console
+ * $ mvn -pl diktat-maven-plugin verify
+ * ```
+ *
+ * from your terminal. If multiple JDKs are installed, be sure to pass
+ * `JAVA_HOME` to the _run configuration_, so that the parent and the forked
+ * JVMs have the same version.
  */
 @OptIn(ExperimentalPathApi::class)
 @MavenJupiterExtension
 class DiktatMavenPluginIntegrationTest {
+    @BeforeEach
+    fun beforeEach(testInfo: TestInfo) {
+        val method = testInfo.testMethod.orElse(null) ?: return
+
+        (Path("target") / "jacoco-it-${method.name}.exec").deleteIfExists()
+    }
+
     @MavenTest
     @MavenGoal("diktat:check@diktat")
-    fun diktatCheck(result: MavenExecutionResult) {
+    fun diktatCheck(testInfo: TestInfo, result: MavenExecutionResult) {
         Assertions.assertEquals(1, result.returnCode)
         Assertions.assertFalse(result.isSuccessful)
         Assertions.assertTrue(result.isFailure)
 
         val mavenLog = result.mavenLog.stdout.readText()
-        Assertions.assertTrue(
-            mavenLog.contains("[FILE_NAME_MATCH_CLASS]")
-        )
 
+        mavenLog.assertContains("[FILE_NAME_MATCH_CLASS]")
+
+        val method = testInfo.testMethod.get()
         File(result.mavenProjectResult.targetProjectDirectory, "target/jacoco-it.exec").copyTo(
-            File("target/jacoco-it-1.exec")
+            File("target/jacoco-it-${method.name}.exec")
         )
     }
 
     @MavenTest
     @MavenGoal("diktat:fix@diktat")
-    fun diktatFix(result: MavenExecutionResult) {
+    fun diktatFix(testInfo: TestInfo, result: MavenExecutionResult) {
         Assertions.assertEquals(1, result.returnCode)
         Assertions.assertFalse(result.isSuccessful)
         Assertions.assertTrue(result.isFailure)
 
         val mavenLog = result.mavenLog.stdout.readText()
 
-        Assertions.assertTrue(
-            mavenLog.contains(Regex("""Original and formatted content differ, writing to [:\w/\\]+Test\.kt\.\.\."""))
-        )
-        Assertions.assertTrue(
-            mavenLog.contains(Regex("There are \\d+ lint errors"))
-        )
-        Assertions.assertTrue(
-            mavenLog.contains("[MISSING_KDOC_TOP_LEVEL]")
-        )
+        mavenLog.assertContains(Regex("""Original and formatted content differ, writing to [:\w/\\]+Test\.kt\.\.\."""))
+        mavenLog.assertContains(Regex("There are \\d+ lint errors"))
+        mavenLog.assertContains("[MISSING_KDOC_TOP_LEVEL]")
 
+        val method = testInfo.testMethod.get()
         File(result.mavenProjectResult.targetProjectDirectory, "target/jacoco-it.exec").copyTo(
-            File("target/jacoco-it-2.exec")
+            File("target/jacoco-it-${method.name}.exec")
         )
     }
+
+    /**
+     * Asserts that this string contains a [substring][other].
+     *
+     * @param other the expected substring.
+     */
+    private inline fun String.assertContains(other: CharSequence,
```

Comment:
I thought that `assertContains` is there in `kotlin.test`. Why do we need these?

Reply:
Indeed. But we don't have it added to our `pom.xml` yet. Should I add `kotlin-test` instead?